ann_benchmarks/data.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import absolute_import
```

Comment:
what is `data.py` used for?

Reply:
It's part of the support code for datasets of different types. `data.py` defines functions for serialising and unserialising each entry in a dataset, and the loader in `ann_benchmarks.datasets` uses its `type_info` map to find these functions. (`ann_benchmarks.algorithms.subprocess`, our mechanism for talking to experiments running in external processes, also uses them to convert entries back into plaintext for transmission.)

Reply:
ok is this the same data format as `protocol`?